io/log_stream.go

```diff
@@ -0,0 +1,41 @@
+package io
+
+import (
+	"context"
+	"io"
+)
+
+type logBody struct {
+	ctx   context.Context
+	f     func([]byte)
+	input io.ReadCloser
+}
```

Comment:
WDYT about 
```go
type ReadCloseInspector struct {
	io.ReadCloser
	AfterRead  func([]byte, error)
	AfterClose func(error)
}

func (rci *ReadCloseInspector) Read(p []byte) (int, error) {
	n, err := rci.ReadCloser.Read(p)
	if n > 0 && rci.AfterRead != nil {
		rci.AfterRead(p[:n], err)
	}
	return n, err
}

func (rci *ReadCloseInspector) Close() error {
	err := rci.ReadCloser.Close()
	if rci.AfterClose != nil {
		rci.AfterClose(err)
	}
	return err
}
```
it can do logging, or e.g. count chunk lengths and log only after Close.

Reply:
This looks wrong for chunked readers

Reply:
I think I have done a good one